config/class-vip-jetpack-network-launch-redirects.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Automattic\VIP\Config;
+
+use Automattic\VIP\Utils\Context;
+
+class VIP_Jetpack_Network_Launch_Redirects {
+	private static $instance;
+
+	const NETWORK_TEMP_REDIRECTS_OPTION_NAME = 'vip_network_temp_redirects';
+	const REDIRECT_TTL_MINUTES               = 60;
+	const LOG_FEATURE_NAME                   = 'vip_network_launch_redirects';
+	// This is used in front of the source domain to avoid the S&R to replace it
+	const URL_REPLACE_PREFIX = '##';
+
+	public static function maybe_redirect_jetpack_network_launches() {
+		// applies redirects only in the frontend and for multisites
+		if ( ! is_multisite() || is_admin() || ! Context::is_web_request() ) {
+			return;
+		}
```

Comment:
`Context::is_web_request()` will fail for xmlrpc requests, which I think it what we're also wanting to target here for JP requests. So may need to get a bit more specific here on what we want to early return on.